oprf/oprf.go

```diff
@@ -0,0 +1,55 @@
+package oprf
+
+import (
+	"github.com/cloudflare/circl/oprf/group"
+)
+
+// BlindToken corresponds to a token that has been blinded.
+// Internally, it is a serialized Point/Element.
+type BlindToken []byte
+
+// IssuedToken corresponds to a token that has been issued.
+// Internally, it is a serialized Point/Element.
+type IssuedToken []byte
+
+// Token is the object issuance of the protocol
+type Token struct {
+	data  []byte
+	blind *group.Scalar
+}
+
+// Evaluation corresponds to the evaluation over a token
+type Evaluation struct {
+	element []byte
+}
+
+// PublicKey is a struct representing a public key
+type PublicKey *group.Point
+
+// PrivateKey is a struct representing a private key
+type PrivateKey *group.Scalar
+
+// Client is
+type Client interface {
+	// SetUp configures a new client by generating its context and keys
+	SetUp(group.CiphersuiteID) (string, PrivateKey)
```

Comment:
Why not `New`?

Reply:
Also, what are the output values? Why does this return a `string` and `PrivateKey`?

Reply:
Copying the definition from the draft:

```
def SetupBaseServer(suite):
  (skS, _) = KeyGen(GG)
  contextString = I2OSP(modeBase, 1) || I2OSP(suite.ID, 2)
  return ServerContext(contextString, skS)
```

It returns a context, which is a string, and the private key.

Reply:
The draft calls it setup and that is what I was following. It can be renamed to new as well.